### PR TITLE
Add compatibility for `view`

### LINKED ISSUE #235: @compat for view
`sub`/`slice` are now replaced with `view`, which isn't in Compat.jl yet

### PR DESCRIPTION
**Pull request by ranjanan**
Test added, README modified. Fixes #235 .

### CODE REVIEW

File: src/Compat.jl [hunk 1]
```diff
@@ -1276,6 +1276,12 @@ else
     end
 end
 
+if isdefined(Base, :view) 
+    nothing
```
Comment (yuyichao) [inline src/Compat.jl]: You should reverse the branch condition and delete this....

File: src/Compat.jl [hunk 2]
```diff
@@ -1276,6 +1276,12 @@ else
...
+else
+    const view = slice
```
Comment (yuyichao) [inline src/Compat.jl]: Is this not exported for `ArrayView` compatibility?

Comment (ranjanan) [inline src/Compat.jl]: Yes.

File: test/runtests.jl [hunk 3]
```diff
@@ -1242,3 +1243,8 @@ end
 @test allunique(1:3)
 @test allunique(FloatRange(0.0, 0.0, 0.0, 1.0))
 @test !allunique(FloatRange(0.0, 0.0, 2.0, 1.0))
+
+# Add test for Base.view
+let a = rand(10,10)
+    @test view(a, 1, :) == slice(a, 1, :)
```
Comment (yuyichao) [inline test/runtests.jl]: Won't this trigger a `depwarn` on 0.5? Maybe construct a `view` and compare it with `getindex` and check that mutating the view aliases the original one.

Comment (ranjanan) [inline test/runtests.jl]: Oh yeah, I think it would. I'll compare with `getindex`

File: test/runtests.jl [hunk 4]
```diff
@@ -1242,3 +1243,8 @@ end
 @test allunique(1:3)
 @test allunique(FloatRange(0.0, 0.0, 0.0, 1.0))
 @test !allunique(FloatRange(0.0, 0.0, 2.0, 1.0))
+
+# Add test for Base.view
+let a = rand(10,10)
+    @test Array(view(a, :, 1)) == a[:,1]
```
Comment (yuyichao) [inline test/runtests.jl]: The `Array` shouldn't be necessary.

Comment (ranjanan) [inline test/runtests.jl]: You are right, it seems to work without that.

File: README.md [hunk 5]
```diff
@@ -194,6 +194,8 @@ Currently, the `@compat` macro supports the following syntaxes:
 
 * `write(::IO, ::Ptr, len)` is now `unsafe_write` [#14766](https://github.com/JuliaLang/julia/pull/14766).
 
+* `slice` is now `view`: do `import Compat.view` and then use `view` normally without the `@compat` macro.
```
Comment (tkelman) [inline README.md]: reference the julia PR - I think it was by `@simonbyrne` if that helps find it faster

Comment (ranjanan) [inline README.md]: Yeah, this: https://github.com/JuliaLang/julia/pull/16972

**Comment by yuyichao** [comment]
LGTM (need squash or squash on merge)

**Comment by ranjanan** [comment]
@tkelman or someone, can we merge, and tag a new release please?

**Comment by ranjanan** [comment]
@TotalVerb can you tag a new release please? There are packages that are dependent on this. Thanks!

**Comment by TotalVerb** [comment]
I will once [CI passes](https://ci.appveyor.com/project/quinnj/compat-jl/branch/master).

**Comment by TotalVerb** [comment]
@tkelman I can't figure out how to tag a release. Could you handle this one?

**Comment by timholy** [comment]
https://github.com/JuliaLang/METADATA.jl/pull/5500

**Pull request merged**